pkgs/by-name/dd/ddsmt/package.nix

```
@@ -0,0 +1,33 @@
+{ lib
+, python3Packages
+, fetchPypi
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "ddsmt";
+  version = "2.0.3";
+  pyproject = true;
+
+  src = fetchPypi {
+    inherit version;
+    pname = "ddSMT";
+    hash = "sha256-nmhEG4sUmgpgRUduVTtwDLGPJVKx+dEaPb+KjFRwV2Q=";
+  };
```

I am not a fan of  `rec`:

```suggestion
let
  pname = "ddsmt";
  version = "2.0.3";
  src = fetchPypi {
    pname = "ddSMT";
    inherit version;
    hash = "sha256-nmhEG4sUmgpgRUduVTtwDLGPJVKx+dEaPb+KjFRwV2Q=";
  };
in 
python3Packages.buildPythonApplication rec {
  inherit pname version src;
  pyproject = true;
```

---

I got rid of the `rec`, but I only hoisted `version` out of the derivation rather than `pname`, `version` and `src`. If that's not alright let me know.